internal/app/init/pkg/system/system.go

```diff
@@ -138,3 +138,28 @@ func (s *singleton) List() (result []*ServiceRunner) {
 
 	return
 }
+
+// Stop will initiate a shutdown of the specified service.
+func (s *singleton) Stop(services ...Service) {
+	if len(services) == 0 {
+		return
+	}
+
+	s.mu.Lock()
+	if s.terminating {
+		s.mu.Unlock()
+		return
+	}
+
+	stateCopy := make(map[string]*ServiceRunner)
+	for _, svc := range services {
+		id := svc.ID(s.UserData)
+		stateCopy[id] = s.State[id]
+	}
+	s.mu.Unlock()
+
+	for id, svc := range stateCopy {
+		svc.Shutdown()
+		WaitForService(StateEventDown, id)
```

Comment:
this just creates a condition, but doesn't wait for it, so you need to add `.Wait(ctx)`

I think this function would benefit from having a context as the first argument, so that it doesn't wait if client aborts (context passed from grpc)

Reply:
plus idk how you feel about it, but you can also call `.Shutdown()` for all of them, and combine all the conditions into one big `WaitForAll` and wait for all of them to shutdown ;)